### PR TITLE
8.0 l10n br account banking payment2

### PR DESCRIPTION
**Pull request by rvalyi**
l10n_br_account_banking_payment extraction + rebase from https://github.com/odoo-brazil/odoo-brazil-banking.git incubator repo

replaces https://github.com/OCA/l10n-brazil/pull/581

### CODE REVIEW

**Comment by mbcosta** [comment]
Hi @mileo @fernandomr @Hendrixcosta 
     There are some commented files and data empty on this module: 
 - l10n_br_account_banking_payment/data/payment_mode_type.xml
 - l10n_br_account_banking_payment/demo/banking_demo.xml ( it's empty, maybe here we can put the same data from payment_mode_type.xml )
 - l10n_br_account_banking_payment/views/bank_payment_manual.xml
 - l10n_br_account_banking_payment/views/payment_mode.xml
 - l10n_br_account_banking_payment/views/payment_mode_type.xml

Because of PEP8 we need to solve this, so what is do you think the better thing to do here, fix or remove ?

**Comment by rvalyi** [comment]
@mbcosta I checked these XML files and I think you can remove them.

**Comment by mbcosta** [comment]
Error in Travis fix in PR https://github.com/OCA/l10n-brazil/pull/595

File: l10n_br_account_banking_payment/models/res_partner_bank.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# @ 2016 Akretion - www.akretion.com.br -
+#   Magno Costa <magno.costa@akretion.com.br>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from openerp import models, api
+
+
+class ResBank(models.Model):
+
```
Comment (rvalyi) [inline l10n_br_account_banking_payment/models/res_partner_bank.py]: please remove empty line

File: l10n_br_account_banking_payment/README.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Brazil Account Banking - Payments Infrastructure
+================================================
+
+This module provide an infrastructure to export payment orders and debit orders
```
Comment (rvalyi) [inline l10n_br_account_banking_payment/README.rst]: s/provide/provides

File: l10n_br_account_banking_payment/models/account_payment.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    @api.multi
+    def action_open(self):
+        """
+        Validacao para nao confirmar ordem de pagamento vazia
+        """
+        for record in self:
+            if not record.line_ids:
+                raise ValidationError(_("Impossible confirm empty line!"))
+        res = super(PaymentOrder, self).action_open()
```
Comment (rvalyi) [inline l10n_br_account_banking_payment/models/account_payment.py]: res variable is useless, better to do return super(...) directly

**Review by rvalyi — COMMENTED** [review]
some minor changes requested

**Review by rvalyi — COMMENTED** [review]
:+1: I technically made the initial commit extraction (so I cannot do a regular approval). But most of the work in this PR is from Kmee and Magno from Akretion so I do approve the PR now that the code is clean enough. Further cleaning may occur during v10 or v11 but let's first get this basic piece merged inside the OCA.

**Comment by mbcosta** [comment]
Thanks @rvalyi 

Hi @fernandomr and @Hendrixcosta please inform your email to include it on README file

cc @mileo

**Comment by mileo** [comment]
mileo@kmee.com.br
hendrix.costa@kmee.com.br
fernando.marcato@kmee.com.br

File: l10n_br_data_base/__openerp__.py [hunk 4]
```diff
@@ -9,6 +9,8 @@
     'version': '8.0.1.0.1',
     'depends': [
         'l10n_br_base',
+        # TODO - https://github.com/OCA/l10n-brazil/issues/588
+        'l10n_br_account_banking_payment',
```
Comment (mileo) [inline l10n_br_data_base/__openerp__.py:13]: Isto é mesmo necessário?

File: l10n_br_account_banking_payment/wizard/payment_order_create_view.xml [hunk 5]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  Copyright (C) 2013 Akretion (http://www.akretion.com)
+  @author: Alexis de Lattre <alexis.delattre@akretion.com>
```
Comment (mileo) [inline l10n_br_account_banking_payment/wizard/payment_order_create_view.xml]: Este copyright esta errado.... @mbcosta 
Apesar de ter sido um cópia do oca/banking não sei se faz sentido deixar esse header ai.

O commit inicial esta aqui: https://github.com/akretion/l10n-brazil/commit/b5274f3397fb544da5f8cbbadb1dbb0573173ff5#diff-c4ae4b617966af609cf36732ad9e0886

**Review by mileo — APPROVED** [review]
Dei uma revisa do no código e esta bem limpo e simples. Vai dar subsidio para os próximos passos.

Obrigado pela extração e o trabalho para realizar o merge aqui.

Vejam só a questão da dependência que eu citei e podem realizar o merge.

File: l10n_br_data_base/__openerp__.py (hunk 4 above)
Comment (rvalyi) [reply l10n_br_data_base/__openerp__.py:13]: @mileo ate onde eu vi sim. Senao teriamos que alterar a validaão do BIC como explicado no issue... Eu acho que na v10 e adiante podemos faze-lo e tirar a dependencia, mas na v8 acho desnecessario gastar tempo nisso...

Comment (mbcosta) [reply l10n_br_data_base/__openerp__.py:13]: ola @mileo , eu prefiro deixar até para lembrar que isso precisa ser resolvido no momento da migração

Comment (mileo) [reply l10n_br_data_base/__openerp__.py:13]: :+1:

**Pull request merged**